crux-lucene/test/crux/lucene_test.clj

```diff
@@ -238,6 +240,46 @@
         (catch Exception t
           (t/is (= "Lucene store latest tx mismatch" (ex-message (ex-cause t)))))))))
 
+(defn ^ICruxAPI start-rocks-lucene-node [{:keys [node-dir index-dir lucene-dir]}]
+  (c/start-node (merge {:crux/document-store {:kv-store {:crux/module `rocks/->kv-store,
+                                                         :db-dir (io/file node-dir "documents")}}
+                        :crux/tx-log {:kv-store {:crux/module `rocks/->kv-store,
+                                                 :db-dir (io/file node-dir "tx-log")}}}
+                       (when lucene-dir
+                         {:crux.lucene/lucene-store {:db-dir lucene-dir}})
+                       (when index-dir
+                         {:crux/index-store {:kv-store {:crux/module `rocks/->kv-store,
+                                                        :db-dir index-dir}}}))))
+
+(t/deftest test-lucene-node-restart
+  (t/testing "test restart with nil indexes"
+    (fix/with-tmp-dir "test-node" [node-dir]
+      (fix/with-tmp-dir "lucene" [lucene-dir]
+        (with-open [node (start-rocks-lucene-node {:node-dir node-dir :lucene-dir lucene-dir})]
+          (submit+await-tx node [[:crux.tx/put {:crux.db/id :ivan :name "Ivan"}]]))
+        (with-open [node (start-rocks-lucene-node {:node-dir node-dir :lucene-dir lucene-dir})]
+          (c/sync node)
+          (t/is (= (c/entity (c/db node) :ivan) {:crux.db/id :ivan :name "Ivan"}))))))
+
+  (t/testing "restart with partial indices (> lucene-tx-id tx-id)"
+    (fix/with-tmp-dir "test-node" [node-dir]
+      (fix/with-tmp-dir "lucene" [lucene-dir]
+        (fix/with-tmp-dir "indices" [index-dir]
+          (with-open [node (start-rocks-lucene-node {:node-dir node-dir :lucene-dir lucene-dir :index-dir index-dir})]
```

Comment:
I'd be tempted to macro this one, you're using the same structure a good few times in this test - let's pair?

Reply:
or see the change @refset and I have made [here](https://github.com/refset/crux/commit/30eee5e52d5f5fb47d7e1388df5277562f5bd9a6#diff-1a4e8b1004a8df52f3b66d07ef537c3e15e561cf037cea7ff44ec4cb1e0250ecR19)